### PR TITLE
Some cleanup in PVT classes

### PR DESCRIPTION
**Pull request by akva2**
- move more code to translation units
- use std::size_t
- cosmetics

also fixes build the multiplexers that i introduced yesterday - we now need the translation units even if ECL_INPUT is not enabled.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for taking on this task.  This looks good to me and I'll merge into master.

File: opm/material/fluidsystems/blackoilpvt/BrineCo2Pvt.cpp [hunk 1]
```diff
@@ -118,8 +144,86 @@ initFromState(const EclipseState& eclState, const Schedule&)
...
+    switch (thermalMixingModelSalt) {
+    case 0: saltMixType_ = Co2StoreConfig::SaltMixingType::NONE; break;
+    case 1: saltMixType_ = Co2StoreConfig::SaltMixingType::MICHAELIDES; break;
+    default: OPM_THROW(std::runtime_error, "The thermal mixing model option for salt are 0 or 1");
+    }
+
+    switch (thermalMixingModelLiquid) {
+    case 0: liquidMixType_ = Co2StoreConfig::LiquidMixingType::NONE; break;
+    case 1: liquidMixType_ = Co2StoreConfig::LiquidMixingType::IDEAL; break;
+    case 2: liquidMixType_ = liquidMixType_ = Co2StoreConfig::LiquidMixingType::DUANSUN;; break;
```
Comment (bska) [inline opm/material/fluidsystems/blackoilpvt/BrineCo2Pvt.cpp:195]: Sorry for not reading this more carefully during review.  This statement (case `2`) appears to assign `liquidMixType_` twice without an intervening sequence point.  Is that intentional?  If not, we should remove one of the references to `liquidMixType_` and/or replace it with the intended object.  Might also want to remove one of the terminating semicolons on `DUANSUN`.

Comment (akva2) [reply opm/material/fluidsystems/blackoilpvt/BrineCo2Pvt.cpp:195]: woops. some shoddy c&p. supposed to be only one liquidMixType_ here, https://github.com/OPM/opm-common/pull/4215

**Pull request merged**